blocks/logic.ts

```diff
@@ -304,100 +314,107 @@ const CONTROLS_IF_MUTATOR_MIXIN = {
   /**
    * Create XML to represent the number of else-if and else inputs.
    * Backwards compatible serialization implementation.
-   * @return {Element} XML storage element.
-   * @this {Block}
+   *
+   * @return XML storage element.
    */
-  mutationToDom: function() {
+  mutationToDom: function(this: IfBlock): Element|null {
     if (!this.elseifCount_ && !this.elseCount_) {
       return null;
     }
     const container = xmlUtils.createElement('mutation');
     if (this.elseifCount_) {
-      container.setAttribute('elseif', this.elseifCount_);
+      container.setAttribute('elseif', String(this.elseifCount_));
     }
     if (this.elseCount_) {
-      container.setAttribute('else', 1);
+      container.setAttribute('else', '1');
     }
     return container;
   },
   /**
    * Parse XML to restore the else-if and else inputs.
    * Backwards compatible serialization implementation.
-   * @param {!Element} xmlElement XML storage element.
-   * @this {Block}
+   *
+   * @param xmlElement XML storage element.
    */
-  domToMutation: function(xmlElement) {
-    this.elseifCount_ = parseInt(xmlElement.getAttribute('elseif'), 10) || 0;
-    this.elseCount_ = parseInt(xmlElement.getAttribute('else'), 10) || 0;
+  domToMutation: function(this: IfBlock, xmlElement: Element) {
+    const elseifAttribute = xmlElement.getAttribute('elseif');
+    if (elseifAttribute === null) {
+      throw new TypeError('xmlElement did not elseif attribute');
+    }
+    this.elseifCount_ = parseInt(elseifAttribute, 10) || 0;
+    this.elseCount_ = parseInt(xmlElement.getAttribute('else')!, 10) || 0;
     this.rebuildShape_();
   },
   /**
    * Returns the state of this block as a JSON serializable object.
-   * @return {?{elseIfCount: (number|undefined), haseElse: (boolean|undefined)}}
-   *     The state of this block, ie the else if count and else state.
+   * @return The state of this block, ie the else if count and else state.
    */
-  saveExtraState: function() {
-    if (!this.elseifCount_ && !this.elseCount_) {
-      return null;
-    }
-    const state = Object.create(null);
-    if (this.elseifCount_) {
-      state['elseIfCount'] = this.elseifCount_;
-    }
-    if (this.elseCount_) {
-      state['hasElse'] = true;
-    }
-    return state;
-  },
+  saveExtraState: function(this: IfBlock): IfExtraState |
+      null {
+        if (!this.elseifCount_ && !this.elseCount_) {
+          return null;
+        }
+        const state = Object.create(null);
+        if (this.elseifCount_) {
+          state['elseIfCount'] = this.elseifCount_;
+        }
+        if (this.elseCount_) {
+          state['hasElse'] = true;
+        }
+        return state;
+      },
```

Comment:
Was this formatting intentional?

Reply:
No.  Probably clang-format being a bit flighty.  We're not prettifying this directory (yet) so I've manually reformatted it to something more reasonable.